lobsterpy/plotting/__init__.py

```diff
@@ -606,71 +588,182 @@ def add_all_relevant_cohps(
+                if label_with_count + suffix not in self._cohps:
+                    self._cohps[label_with_count + suffix] = {}
                 for label in labels:
                     cohp = complete_cohp.get_cohp_by_label(label)
                     energies = (
                         cohp.energies - cohp.efermi
                         if self.zero_at_efermi
                         else cohp.energies
                     )
-                    outer_key = label_with_count + suffix
-                    struct = analyse.structure
-                    atom_pairs = []
-                    for site in complete_cohp.bonds[label]["sites"]:
-                        atom = site.species_string + str(struct.sites.index(site) + 1)
-                        atom_pairs.append(atom)
-                    key = "{}: {} ({} \u00c5)".format(
-                        label,
-                        "-".join(atom_pairs),
-                        str(round(complete_cohp.bonds[label]["length"], 2)),
+                    drop_down_key = label_with_count + suffix
+                    plot_legend = self._get_plot_lable_for_label_resolved(
+                        structure=analyse.structure,
+                        label_list=[label],
+                        complete_cohp=complete_cohp,
+                        orb_list=[],
+                        label_resolved=True,
+                        orbital_resolved=False,
                     )
-                    self._cohps[outer_key].update(
+                    if len(plot_data) > 1:
+                        self._cohps[drop_down_key].update(
+                            {
+                                plot_legend: {
+                                    "energies": energies,
+                                    "COHP": cohp.get_cohp(),
+                                    "ICOHP": cohp.get_icohp(),
+                                    "efermi": cohp.efermi,
+                                }
+                            }
+                        )
+
+                    plot_legend_here = plot_legend + suffix
+                    self._cohps["All"].update(
                         {
-                            key: {
+                            plot_legend_here: {
                                 "energies": energies,
                                 "COHP": cohp.get_cohp(),
                                 "ICOHP": cohp.get_icohp(),
                                 "efermi": cohp.efermi,
                             }
                         }
                     )
+            # will add cohp data for each relevant orbitals and bond label iteratively
+            if orbital_resolved and label_resolved:
+                if label_with_count + suffix not in self._cohps:
+                    self._cohps[label_with_count + suffix] = {}
+                plot_data_orb = analyse.get_site_orbital_resolved_labels()
+                drop_down_key = label_with_count + suffix
```

Comment:
this looks similar below.

Reply:
> this looks similar below.

Now with comments added, it probably makes clearer how it differ